### PR TITLE
feat: ForcedPolicyRedemption model

### PR DESCRIPTION
**Pull request by iloveagent57**
There is frequently a need to force through a redemption (and related enrollment/fulfillment) of a particular learner, covered by a particular subsidy access policy, into some specific course run. This commit introduces a new Django Model to support such an operation. ENT-8703

Related ADR: https://github.com/openedx/enterprise-access/pull/427

### CODE REVIEW

File: enterprise_access/apps/content_assignments/utils.py [hunk 1]
```diff
@@ -1,17 +0,0 @@
-"""
```
Comment (iloveagent57) [inline enterprise_access/apps/content_assignments/utils.py:1]: Moved to `enterprise_access.apps.utils`

File: enterprise_access/apps/content_assignments/tests/factories.py [hunk 2]
```diff
@@ -8,21 +8,11 @@
 from faker import Faker
 
 from ..models import AssignmentConfiguration, LearnerContentAssignment
+from test_utils import random_content_key
 
 FAKER = Faker()
 
 
-def random_content_key():
```
Comment (iloveagent57) [inline enterprise_access/apps/content_assignments/tests/factories.py:15]: moved to top level `test_utils`

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -12,7 +12,7 @@ jobs:
     strategy:
       matrix:
         python-version: ["3.8"]
-        django-version: ["pinned", "4.2"]
+        django-version: ["pinned"]
```
Comment (iloveagent57) [inline .github/workflows/ci.yml:15]: The pinned version _is_ 4.2, no reason to run these twice.

File: enterprise_access/apps/subsidy_access_policy/models.py [hunk 4]
```diff
@@ -1509,3 +1511,148 @@ class Meta:
...
+        return (
+            f'policy_uuid={self.subsidy_access_policy.uuid}, transaction_uuid={self.transaction_uuid}, '
+            f'lms_user_id={self.lms_user_id}, course_run_key={self.course_run_key}'
+        )
+
+    def create_assignment(self):
+        """
+        For assignment-based policies, an allocated ``LearnerContentAssignment`` must exist
+        before redemption can occur.
+        """
```
Comment (pwnage101) [inline enterprise_access/apps/subsidy_access_policy/models.py:1603]: Not entirely sure what the plan is for assignments, but I assume this logic will change later to support assigning course run keys?

**Review by pwnage101 — APPROVED** [review]

File: enterprise_access/apps/subsidy_access_policy/models.py (hunk 4 above)
Comment (pwnage101) [reply enterprise_access/apps/subsidy_access_policy/models.py:1603]: Oh, I should probably read the ADR later...

**Pull request merged**